Feature 721 (#780)
italia/design-react-kit: stories/Tables/Tables.stories.mdx

import { Meta, Canvas, Story } from '@storybook/addon-docs';
import { Callout, CalloutText, CalloutTitle } from '../../src';

<Meta title='Documentazione/Organizzare i contenuti/Tabelle' />

# Tabelle

Documentazione ed esempi per lo stile delle tabelle.

## Esempi

> A causa dell’uso diffuso di tabelle su widget di terze parti come calendari e campi di selezione data, Bootstrap ha progettato le tabelle senza forzarne lo stile. È sufficiente aggiungere la classe `.table` a qualsiasi `<table>`, quindi estendere con stili personalizzati o con le nostre varie classi incluse di modificatori.

Author: Effectively. Removed

Usando il markup di base della tabella, ecco come appaiono le tabelle che utilizzano la classe `.table` in Bootstrap. **Tutti gli stili di tabella sono ereditari**, il che significa che tutte le tabelle nidificate saranno stilizzate nello stesso modo del genitore.

<Canvas>
  <Story id='componenti-tables--table-base' />
</Canvas>

Puoi anche invertire i colori, con testo chiaro su sfondi scuri con `.table-dark`.

<Canvas>
  <Story id='componenti-tables--table-dark' />
</Canvas>

## Opzioni delle intestazioni di tabella

Simile alle tabelle e alle tabelle scure, utilizzare le classi di modifica `.thead-light` o `.thead-dark` per far apparire `<thead>` chiaro o grigio scuro.

<Canvas>
  <Story id='componenti-tables--table-thead' />
</Canvas>

## Righe striate

Usa `.table-striped` per aggiungere delle striature zebrate ad ogni riga della tabella contenute in `<tbody>`.

<Canvas>
  <Story id='componenti-tables--table-striped' />
</Canvas>

## Tabella con bordi

Aggiungi `.table-bordered` per avere i bordi a tutti i lati della tabella e su tutte le celle.

<Canvas>
  <Story id='componenti-tables--table-bordered' />
</Canvas>

## Tabella senza bordi

> Aggiungi la classe `.table-borderless` per una tabella senza bordi.

Reviewer: ```suggestion
Usa l'attributo `borderless` per una tabella senza bordi.
```

<Canvas>
  <Story id='componenti-tables--table-borderless' />
</Canvas>

## Righe ed hover

> Aggiungi `.table-hover` per abilitare lo stato hover sulle righe della tabella contenute in `<tbody>`.

Reviewer: ```suggestion
Usa l'attributo `hover` per abilitare lo stato hover sulle righe della tabella contenute in `<tbody>`.
```

<Canvas>
  <Story id='componenti-tables--table-hover' />
</Canvas>

## Tabella compatta

> Aggiungi `.table-sm` per rendere le tabelle più compatte dimezzando il cell padding.

Reviewer: ```suggestion
Aggiungi l'attributo `size="sm"` per rendere le tabelle più compatte dimezzando il cell padding.
```

    <CalloutText>
      L'uso del colore per aggiungere un significato fornisce solo
      un’indicazione visiva, che non sarà trasmesso agli utenti di tecnologie
      assistive – come gli screen reader. Assicurati che le informazioni
      denotate dal colore siano rese disponibili anche dal contenuto stesso
      (es.: il testo visibile), o siano incluse attraverso mezzi alternativi,
      come testo aggiuntivo nascosto con la classe `.sr-only`.
    </CalloutText>
  </Callout>
</div>

Crea tabelle responsive racchiudendo ogni `.table` con `.table-responsive{-sm|-md|-lg|-xl}`, facendo scorrere orizzontalmente la tabella ad ogni breakpoint, rispettivamente con `max-width` di 576px, 768px, 992px, 1200px.

<div className='docs'>
  <Callout color='primary'>
    <CalloutText>
      Nota che in quanto i browser non supportano attualmente i range context
      queries, stiamo aggirando i limiti dei prefissi min- and max- e viewports
      con larghezze frazionarie (che possono verificarsi in determinate
      condizioni su dispositivi ad alta risoluzione, ad esempio) utilizzando
      valori con maggiore precisione per questi confronti.
    </CalloutText>
  </Callout>

> </div>

Reviewer: I believe this content belongs to the `Responsive` section below. Opened an issue also on BI: https://github.com/italia/bootstrap-italia/issues/566

Author: I have removed paragraph with refs of responsive classname and moved the callout in correct section. What do you think?

## Captions

Un `<caption>` funziona come un’intestazione per una tabella. Aiuta gli utenti con screen reader a trovare una tabella e capire di cosa si tratta e decidere se vogliono leggerla.

<Canvas>
  <Story id='componenti-tables--table-captions' />
</Canvas>

# Tabelle responsive

> Le tabelle responsive consentono di scorrere le tabelle orizzontalmente con facilità. Rendi ogni tabella responsive su tutti i viewports racchiudendo un `.table` con `.table-responsive`. Oppure, scegli un breakpoint massimo con il quale ottenere una tabella responsive usando `.table-responsive{-sm|-md|-lg|-xl}`.

Reviewer: ```suggestion
Le tabelle responsive consentono di scorrere le tabelle orizzontalmente con facilità. Rendi ogni tabella responsive su tutti i viewports aggiungendo al componente `Table` l'attributo `responsive`. Oppure, scegli un breakpoint massimo con il quale ottenere una tabella responsive combinando gli attributi `responsive` e `size`.
```

<div className='docs'>
  <Callout color='primary'>
    <CalloutTitle>Ritaglio / troncamento verticale</CalloutTitle>
    <CalloutText>
      Le tabelle responsive fanno uso di `overflow-y: hidden`, che rimuove
      qualsiasi contenuto che va oltre i bordi inferiore o superiore della
      tabella. In particolare, questo può ritagliare i menu a discesa e altri
      widget di terze parti.
    </CalloutText>
  </Callout>
</div>

## Sempre responsive

> Attraverso ogni breakpoint, usa `.table-responsive` per tabelle con scorrimento verticale.

Reviewer: ```suggestion
Attraverso ogni breakpoint, usa l'attributo `responsive` per tabelle con scorrimento verticale.
```

<Canvas>
  <Story id='componenti-tables--table-responsive' />
</Canvas>

## Breakpoint specifici

> Usa `.table-responsive{-sm|-md|-lg|-xl}` come necessario per creare tabelle responsive fino a un punto di interruzione particolare. Da quel punto di interruzione in su, la tabella si comporterà normalmente e non scorrerà orizzontalmente.

Reviewer: ```suggestion
Combina gli attributi `responsive` e `size` come necessario per creare tabelle responsive fino a un punto di interruzione particolare. Da quel punto di interruzione in su, la tabella si comporterà normalmente e non scorrerà orizzontalmente.
```
